### PR TITLE
BigtableIO: Force minimum value of `desiredBundleSizeBytes` to be 1

### LINKED ISSUE #28793: [Bug]: BigtableSource "Desired bundle size 0 bytes must be greater than 0"
### What happened?

org.apache.beam.runners.direct.BoundedReadEvaluatorFactory.InputProvider#getInitialInputs
org.apache.beam.sdk.io.gcp.bigtable.BigtableIO.BigtableSource#split

Imagine a case where in:
https://github.com/apache/beam/blob/282d027f16214c9581b1d19073252425cff53ee9/runners/direct-java/src/main/java/org/apache/beam/runners/direct/BoundedReadEvaluatorFactory.java#L215-L217
- `targetParallelism` is `32`
- `source.getEstimatedByteSize()` is `10`

then

- `bytesPerBundle` will be `0`

so

https://github.com/apache/beam/blob/282d027f16214c9581b1d19073252425cff53ee9/runners/direct-java/src/main/java/org/apache/beam/runners/direct/BoundedReadEvaluatorFactory.java#L217
will be called with the values: `split.source(0L, options)`

In `OffsetBasedSource#split`, this desired-0-sized split is handled:
https://github.com/apache/beam/blob/282d027f16214c9581b1d19073252425cff53ee9/sdks/java/core/src/main/java/org/apache/beam/sdk/io/OffsetBasedSource.java#L115-L116

But `BigtableSource#split` does not seem to handle the desired-0-sized split:

https://github.com/apache/beam/blob/282d027f16214c9581b1d19073252425cff53ee9/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java#L1328-L1333

so a few frames down the road from `BigtableSource#split` you'll end up violating this `checkArgument`:

https://github.com/apache/beam/blob/71c8459633ec86e576eca080a26be9f42474ecb2/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java#L1623-L1626

### Issue Priority

Priority: 2 (default / most bugs should be filed as P2)

### Issue Components

- [ ] Component: Python SDK
- [X] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam YAML
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by joar**
Fixes #28793 in the way suggested in https://github.com/apache/beam/issues/28793#issuecomment-1745361478:

- `BoundedReadEvaluatorFactory#getInitialInputs` may still calculate a `bytesPerBundle` of `0` (e.g. when `targetParallelism > estimatedBytes`); but
- `BigtableSource#split` will interpret it as `1` in order to not violate the `checkArgument()` in https://github.com/apache/beam/blob/71c8459633ec86e576eca080a26be9f42474ecb2/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java#L1623-L1626

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [x] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -86,6 +86,7 @@
 * Fixed exception chaining issue in GCS connector (Python) ([#26769](https://github.com/apache/beam/issues/26769#issuecomment-1700422615)).
 * Fixed streaming inserts exception handling, GoogleAPICallErrors are now retried according to retry strategy and routed to failed rows where appropriate rather than causing a pipeline error (Python) ([#21080](https://github.com/apache/beam/issues/21080)).
 * Fixed a bug in Python SDK's cross-language Bigtable sink that mishandled records that don't have an explicit timestamp set: [#28632](https://github.com/apache/beam/issues/28632).
+* Fixed "Desired bundle size 0 bytes must be greater than 0" in BigtableSource when you have more cores than bytes to read [#28793](https://github.com/apache/beam/issues/28793).
```
Comment (Abacn) [inline CHANGES.md:89]: This will go to v2.52.0 as v2.51.0 has already been cut

**Review by bvolpato — APPROVED** [review]
Changes LGTM, just +1 on moving the CHANGES.md delta to 2.52.0 since 2.51.0 is almost out / at release candidate fix.

File: CHANGES.md (hunk 1 above)
Comment (Abacn) [reply CHANGES.md:89]: If not sure how to add section, we can just remove this entry for now - will update later

Comment (joar) [reply CHANGES.md]: There's a template at the top of the file, so I used that to create a new section for 2.52 and moved the change note!

**Comment by joar** [comment]
The ["PreCommit Whitespace" failure](https://github.com/apache/beam/actions/runs/6419036054/job/17428039085?pr=28798) seems to me to be unrelated to the changes in this PR:

```
> Task :sdks:python:sdist

FAILURE: Build failed with an exception.

* What went wrong:
A problem was found with the configuration of task ':sdks:python:test-suites:tox:py38:archiveFilesToLint' (type 'Zip').
  - Gradle detected a problem with the following location: '/runner/_work/beam/beam'.
    
    Reason: Task ':sdks:python:test-suites:tox:py38:archiveFilesToLint' uses this output of task ':sdks:python:setupVirtualenv' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed.
    
    Possible solutions:
      1. Declare task ':sdks:python:setupVirtualenv' as an input of ':sdks:python:test-suites:tox:py38:archiveFilesToLint'.
      2. Declare an explicit dependency on ':sdks:python:setupVirtualenv' from ':sdks:python:test-suites:tox:py38:archiveFilesToLint' using Task#dependsOn.
      3. Declare an explicit dependency on ':sdks:python:setupVirtualenv' from ':sdks:python:test-suites:tox:py38:archiveFilesToLint' using Task#mustRunAfter.
    
    For more information, please refer to https://docs.gradle.org/8.3/userguide/validation_problems.html#implicit_dependency in the Gradle documentation.
```

**Comment by joar** [comment]
Running `whitespacelint` manually makes me believe that it would pass if the check was working :)

```console
$ python -m pip install --user whitespacelint==1.1.0
Requirement already satisfied: whitespacelint==1.1.0 in /home/joar/.local/lib/python3.8/site-packages (1.1.0)
$ whitespacelint CHANGES.md
# exits with 0
$ git rev-parse HEAD
3096265e05ff59ae192cc7f4a571e9247937211b
```

**Comment by Abacn** [comment]
whitespace issue was an issue on master and now fixed #28845

**Pull request merged**